nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/SessionImpl.java

```diff
@@ -966,17 +966,21 @@ public void removeNode(Node node) {
         // find all descendants
         List<NodeInfo> nodeInfos = context.getNodeAndDescendantsInfo(node.getHierFragment());
 
-        // check that there is no active retention
-        Set<Serializable> retentionActiveIds = nodeInfos.stream() //
-                                                        .filter(info -> info.isRetentionActive)
-                                                        .map(info -> info.id)
-                                                        .collect(Collectors.toSet());
-        if (!retentionActiveIds.isEmpty()) {
-            if (retentionActiveIds.contains(id)) {
-                throw new DocumentExistsException("Cannot remove " + id + ", it is under active retention");
-            } else {
-                throw new DocumentExistsException("Cannot remove " + id + ", subdocument "
-                        + retentionActiveIds.iterator().next() + " is under active retention");
+        // check that there is no retention / hold
+        Set<Serializable> undeletableIds = nodeInfos.stream() //
+                                                    .filter(info -> info.isUndeletable)
+                                                    .map(info -> info.id)
+                                                    .collect(Collectors.toSet());
+        if (!undeletableIds.isEmpty()) {
+            // in tests we may want to delete everything
+            boolean allowDeleteUndeletable = Framework.isBooleanPropertyTrue("allowDeleteUndeletableDocuments");
```

Comment:
same about replacing it by `Framework.isTestModeSet() && Framework.isBooleanPropertyTrue("allowDeleteUndeletableDocuments")`